py-polars/polars/dataframe/frame.py

```diff
@@ -6865,10 +6865,13 @@ def partition_by(
         ]
 
         if as_dict:
+            df = self._from_pydf(self._df)
             if len(by) == 1:
```

Comment:
Could we do the `unique` call only when `include_key` is `False`?

Reply:
@ritchie you're right, just pushed a non-unique line when `include_key` is `False`.